File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Data/System.Data.IDataReader/IDataReader.GetValueAs.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Data

Public Module Extensions_674

	''' <summary>
	'''     An IDataReader extension method that gets value as.
	''' </summary>
	''' <typeparam name="T">Generic type parameter.</typeparam>
	''' <param name="this">The @this to act on.</param>
	''' <param name="index">Zero-based index of the.</param>
	''' <returns>The value as.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function GetValueAs(Of T)(this As IDataReader, index As Integer) As T
		Return DirectCast(this.GetValue(index), T)
	End Function

	''' <summary>
	'''     An IDataReader extension method that gets value as.
	''' </summary>
	''' <typeparam name="T">Generic type parameter.</typeparam>
	''' <param name="this">The @this to act on.</param>
	''' <param name="columnName">Name of the column.</param>
	''' <returns>The value as.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function GetValueAs(Of T)(this As IDataReader, columnName As String) As T
		Return DirectCast(this.GetValue(this.GetOrdinal(columnName)), T)
	End Function
End Module

File: src/Z.Data/System.Data.IDataReader/IDataReader.GetValueAs.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System.Data;

public static partial class Extensions
{
    /// <summary>
    ///     An IDataReader extension method that gets value as.
    /// </summary>
    /// <typeparam name="T">Generic type parameter.</typeparam>
    /// <param name="this">The @this to act on.</param>
    /// <param name="index">Zero-based index of the.</param>
    /// <returns>The value as.</returns>
    public static T GetValueAs<T>(this IDataReader @this, int index)
    {
        return (T) @this.GetValue(index);
    }

    /// <summary>
    ///     An IDataReader extension method that gets value as.
    /// </summary>
    /// <typeparam name="T">Generic type parameter.</typeparam>
    /// <param name="this">The @this to act on.</param>
    /// <param name="columnName">Name of the column.</param>
    /// <returns>The value as.</returns>
    public static T GetValueAs<T>(this IDataReader @this, string columnName)
    {
        return (T) @this.GetValue(@this.GetOrdinal(columnName));
    }
}